### PR TITLE
Simple camera and basic file restructure

### PR DESCRIPTION
**Pull request by cooltrain7**
- Setup a basic boilerplate file structure for the resources
- Apply the default gd style guide
- Setup ground floor plane with using UV texture
- Implements a simple flying camera

### CODE REVIEW

File: editor/script/camera_controller.gd [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+		Input.get_action_strength("camera_backwards") - Input.get_action_strength("camera_forward"),
+	)
+
+	if(direction != Vector3.ZERO):
+		direction = direction.normalized()
+		var finalSpeed = base_cam_speed * (Input.get_action_strength("camera_boost") +1)
+		translate(direction * finalSpeed * delta)
+
+#Handle input for rotation
+func camera_rotate(delta) -> void:
```
Comment (TheAirMike) [inline editor/script/camera_controller.gd]: To be consistent with `camera_movement` let's name it `camera_rotation`.

Or maybe rename `camera_movement` to `camera_move`.

File: editor/script/camera_controller.gd [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+	var rot_input = Input.get_action_strength("camera_rot_left") - Input.get_action_strength("camera_rot_right")
+	rotate_y(rot_input * base_cam_rot_speed * delta)
```
Comment (TheAirMike) [inline editor/script/camera_controller.gd:26]: It took me some time to figure out exactly which buttons rotated the camera, so we should consider adding a corner text with the button mappings later on.

**Review by TheAirMike — CHANGES_REQUESTED** [review]
Looks good, just 1 teeny-tiny nit-pick

File: editor/script/camera_controller.gd (hunk 1 above)
Comment (cooltrain7) [reply editor/script/camera_controller.gd]: Done.

File: editor/script/camera_controller.gd (hunk 2 above)
Comment (cooltrain7) [reply editor/script/camera_controller.gd:26]: Yea could do it in static text for now but I might just create an UI/options menu branch and do it on that to view controls.

**Review by TheAirMike — APPROVED** [review]

**Pull request merged**